types/react-adaptive-hooks/media-capabilities.d.ts

```diff
@@ -0,0 +1,32 @@
+export interface AudioMediaConfiguration {
+    contentType: string;
+    channels: number;
+    bitrate: number;
+    samplerate: number;
+}
+
+export interface VideoMediaConfiguration {
+    contentType: string;
+    width: number;
+    height: number;
+    bitrate: number;
+    framerate: number;
+}
+
+export type MediaConfiguration = {
+    type: 'file' | 'record' | 'transmission' | 'media-source';
+} & ({ audio: AudioMediaConfiguration } | { video: VideoMediaConfiguration });
+
+export interface MediaConfigurationInfo {
+    supported: boolean;
+    smooth: boolean;
+    powerEfficient: boolean;
+}
+
+export function useMediaCapabilities<T = Partial<MediaConfigurationInfo>>(
+    mediaConfig: MediaConfiguration,
+    initialMediaCapabilities: T,
+): {
+    mediaCapabilities:
+        | ({ supported: boolean; hasMediaConfig: boolean } & T) | ({ hasMediaConfig?: undefined} & MediaConfigurationInfo);
```

Comment:
this seems weird. is the aim that if `hasMediaConfig` is true, you have media configuration info? you should be able to do similar to your `unsupported` types here if so

Reply:
This whole hook is pretty rough. The code isn't even correct yet but has been released. 👎 Not entirely sure what the point of it is but it's in the code.